docs/_specs/service-manager/service-manager.md

```diff
@@ -0,0 +1,127 @@
+
+The following CRUD operations will be supported:
+
+* CREATE - register a service class from a specific broker and make it available for the organization
+* GET - return a service class object including available plans (and possibly schema)
+* LIST - return all service classes
+* DELETE - remove a service class making it unavailable to the organization
+
+### Service Instance
+
+A service instance represents a provisioning or instance of a service.  For instance the instantiation of a postgres
+service class would result in a provisioned database and a binding created (credentials stored as secrets).  The service
+is identifieable with a unique name (e.g. `my-first-postgres`).  Functions can then be registered with services by name
+(same was as secrets) and have the binding data injected into the function context (again, just like secrets).  In fact,
+in this scenario the credentials are stored as secrets and the service instance maintains the secret name, which is how
+the credentials are looked up.
+
+The following CRUD operations will be supported:
+
+* CREATE - create a service instance from a service class
+    - schema will be service class specific
+* GET - return the service which includes the binding information (e.g. Dispatch secret keys)
+* LIST - return a list of all service instances
+* DELETE - delete the service and do any service cleanup
+
+## Design
+
+As stated earlier, the service-manager is a light API layer on top of the kubernetes service catalog.  Additionally,
+we are adding per-organization management of available services.
+
+### Broker object schema
+
+* `name` - `string` - broker name
+* `URL` - `string` - URL to OSBAPI compatible broker
+* `parameters` - `map` - key/value paramteters passed as args to the broker
+* `serviceClasses` - `map` - key/serviceClass mapping (not sure if map or list)
+
+> NOTE: Broker management will not be available immediately.
+
+### Service Class object schema
+
+* `name` - `string` - service class name
+* `brokerID` - `string` - ID of broker
+* `plans` - `map` - key/servicePlan mapping
+    - service plan objects may contain schema with regards to service creation and binding
+
+### Service Instance object schema
+
+* `name` - `string` - service instance name
+* `serviceClassID` - `string` - ID of service class
+* `planID` - `string` - ID of service plan
+* `parameters` - `map` - parameters used for service provisioning and binding
+* `binding` - `map` - binding data (secret names) used by functions to use to this service instance
+
+## Milestones
+
+### Phase 1
+
+* Brokers
+    - no Dispatch broker objects/APIs
+    - registered externally with kubernetes/service-catalog
+* Service Classes
+    - pulled directly from service-catalog (auto-registered)
+    - read-only API (only GET/LIST operations)
+* Service Instance
+    - CREATE/GET/LIST/DELETE all supported
+    - pass `--service <serviceName>` to function to inject binding data
```

Comment:
Just for clarification, we will be supporting more than one service right? Say by specifying --service flag multiple times.